### PR TITLE
Generate verse markers on compiling a chapter

### PR DESCRIPTION
**Pull request by AnonymousWalker**
Generate markers when compiling a chapter from verses. Each marker placed at the beginning of each verse take

---
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/467)

### CODE REVIEW

**Review by mXaln — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/467)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jsarabia)

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/467)* status: all files reviewed, 1 unresolved discussion (waiting on @AnonymousWalker)

___
*[common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/ConcatenateAudio.kt, line 59 at r1](https://reviewable.io/reviews/bible-translation-tools/orature/467#-Ms1HvO6DjG7aF-_NP4E:-Ms1HvO6DjG7aF-_NP4F:b-63ridn) ([raw file](https://github.com/bible-translation-tools/orature/blob/528c0ad31bd53e97aafb5497ac79f5c535d5c592/common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/ConcatenateAudio.kt#L59)):*
> ```Kotlin
>                     )
>                     markerLocation += audioFile.totalFrames
>                     markerId++
> ```

You can't rely on the assumption that verse numbers are sequential. This is incorrect for verse bridges and chunks.

You need to add the starting marker in each of the verse by verse recordings, then in this usecase, read the marker metadata and use the label from the file.

**Review by jsarabia — COMMENTED** [review]
When you've made the correction, please also add unit tests.

> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/467#-:-Ms1IIaLCAhMXFWV-Hvn:bmisypv)* status: all files reviewed, 1 unresolved discussion (waiting on @AnonymousWalker)

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/467)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @AnonymousWalker, @jsarabia, and @mXaln)

___
*[common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/audio/LabeledAudio.kt, line 23 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/467#-MsB9GNS9Y3OZawKcfIj:-MsB9GNS9Y3OZawKcfIk:b3gil5l) ([raw file](https://github.com/bible-translation-tools/orature/blob/1327ec11fa6de0cf944e8c6a2305087886e6812f/common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/audio/LabeledAudio.kt#L23)):*
> ```Kotlin
> import java.io.File
> 
> data class LabeledAudio(val title: String, val file: File)
> ```

I don't want an entire new class just to solve one problem. Otherwise we end up never using the same class for anything, never knowing what something is or why, because everything takes its own slight modification of the same thing.
___
*[common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/ConcatenateAudio.kt, line 64 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/467#-MsB9dyO9dzfjnzNTJ_V:-MsB9dyO9dzfjnzNTJ_W:b-dk82nl) ([raw file](https://github.com/bible-translation-tools/orature/blob/1327ec11fa6de0cf944e8c6a2305087886e6812f/common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/ConcatenateAudio.kt#L64)):*
> ```Kotlin
> 
>                 audioList.forEach { audio ->
>                     val audioFile = AudioFile(audio.file)
> ```

If you have an audio file, you can get its metadata. If you can get its metadata, you can get its list of verses. Use those, not "audio.title"

Place the start marker automatically at the beginning of verse files when they're created, and then the solution here is easy.

**Review by AnonymousWalker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/467)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @jsarabia and @mXaln)

___
*[common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/ConcatenateAudio.kt, line 64 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/467#-MsB9dyO9dzfjnzNTJ_V:-MsfJ4bg3XFm2BBm6Sw4:b-896fix) ([raw file](https://github.com/bible-translation-tools/orature/blob/1327ec11fa6de0cf944e8c6a2305087886e6812f/common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/ConcatenateAudio.kt#L64)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

If you have an audio file, you can get its metadata. If you can get its metadata, you can get its list of verses. Use those, not "audio.title"

Place the start marker automatically at the beginning of verse files when they're created, and then the solution here is easy.
</blockquote></details>

Done.
___
*[common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/audio/LabeledAudio.kt, line 23 at r2](https://reviewable.io/reviews/bible-translation-tools/orature/467#-MsB9GNS9Y3OZawKcfIj:-MsfJ5VA-6zMed1mOrar:b-896fix) ([raw file](https://github.com/bible-translation-tools/orature/blob/1327ec11fa6de0cf944e8c6a2305087886e6812f/common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/audio/LabeledAudio.kt#L23)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

I don't want an entire new class just to solve one problem. Otherwise we end up never using the same class for anything, never knowing what something is or why, because everything takes its own slight modification of the same thing.
</blockquote></details>

Done.

File: jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/ChapterPageViewModel.kt [hunk 1]
```diff
@@ -428,6 +430,15 @@ class ChapterPageViewModel : ViewModel() {
...
 
+    private fun setMarker(marker: String, take: Take) {
+        AudioFile(take.file).apply {
+            if (metadata.getCues().isEmpty()) {
+                metadata.addCue(0, marker)
+                update()
+            }
+        }
+    }
+
```
Comment (AnonymousWalker) [inline jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/ChapterPageViewModel.kt:441]: apply to existing takes which have no marker in metadata

**Review by jsarabia — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/467#-:-MsffmgU7oVlAakoaquy:bnfp4nl)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @jsarabia and @mXaln)

**Pull request merged**